test/Mono.Linker.Tests.Cases/DataFlow/VirtualMethodHierarchyDataflowAnnotationValidation.cs

```diff
@@ -629,22 +629,27 @@ static virtual Type VirtualMethod
 
 			class ImplIDamOnAllMissing : IDamOnAll
 			{
-				[ExpectedWarning ("IL2092")]
-				[ExpectedWarning ("IL2093")]
-				[ExpectedWarning ("IL2095")]
+				// NativeAOT doesn't validate overrides when accessed through reflection because it's a direct call (non-virtual)
```

Comment:
I thought about it, but it's relatively complex to do in the linker and I don't see much value in it.
In general I agree that we should try to make the behavior between linker and AOT as consistent as possible. That said, the trimming behavior between the two tools is going to be different. Linker can't perform the same optimizations Native AOT can (since it has to produce valid IL and has other limitations).  And since diagnostics are only produced for kept code, there will always be differences.

I think the goal should be:
* All tools must produce diagnostics in case there are potential runtime behavioral differences in the app (cover all the holes in the analysis)
* Tools should try to reduce any additional warnings which don't have a direct impact on behavior of the app (basically noise).

If the tools can reduce the noise to 0, then their behavior should be very close, but there will still be difference due to differences in trimming behavior.

So for me, this specific case is linker being a bit too noisy... but it's a niche case and so I see it as a very low priority to do anything about.